apis/python/src/tiledbsoma/_read_iters.py

```diff
@@ -5,59 +5,424 @@
+            )
+            if reindex_disable is not None
+            else []
+        )
+
+        # Currently, only support chunking on one dimension.
+        if len(axis) != 1:
+            raise NotImplementedError(
+                "Multi-dimension blockwise iterators not implemented"
+            )
+        # all dim indices must be in acceptable range
+        if not all(0 <= d < ndim for d in axis):
+            raise ValueError("blockwise `axis` value must be in range [0, ndim)")
+        if not all(0 <= d < ndim for d in reindex_disable):
+            raise ValueError(
+                "blockwise `reindex_disable` value must be in range [0, ndim)"
+            )
+
+        # if not specified, set default size for each axis. Default heuristic
+        # assumes 2D array has many more rows than cols (i.e., n_obs>>n_vars).
+        default_block_size = (2**16,) + (2**8,) * (ndim - 1)
+        if size is None:
+            size = [default_block_size[d] for d in axis]
+        elif isinstance(size, int):
+            size = [size] * len(axis)
+        elif isinstance(size, Sequence):
+            size = list(size) + [default_block_size[d] for d in axis[len(size) :]]
+        else:
+            raise TypeError(
+                "blockwise iterator `size` must be None, int or Sequence[int]"
+            )
+
+        return axis, size, reindex_disable
+
+    @abc.abstractmethod
+    def _create_reader(self) -> Iterator[_RT]:
+        """Sub-class responsibility"""
+        raise NotImplementedError()
+
+    def __next__(self) -> _RT:
+        return next(self._reader)
+
+    def concat(self) -> _RT:
+        """
+        Unimplemented as there is little utility beyond that offered by a ragged
+        read iterator concat, other than reindexing.
+        """
+        raise NotImplementedError("Blockwise iterators do not support concat operation")
+
+    _EagerRT = TypeVar("_EagerRT")
+
+    def _if_eager(
+        self, x: Iterator[_EagerRT], _pool: Optional[ThreadPoolExecutor] = None
+    ) -> Iterator[_EagerRT]:
+        """Private"""
+        return EagerIterator(x, pool=_pool) if self.eager else x
 
+    def _table_reader(self) -> BlockwiseSingleAxisTableIter:
+        """Private. Blockwise table reader. Helper function for sub-class use"""
+        kwargs: Dict[str, object] = {"result_order": self.sr.result_order}
+        for coord_chunk in _coords_strider(
+            self.coords[self.major_axis],
+            self.sr.shape[self.major_axis],
+            self.size[0],
+        ):
+            self.sr.reset(**kwargs)
+            step_coords = [*self.coords]
+            step_coords[self.major_axis] = coord_chunk
+            self.array._set_reader_coords(self.sr, step_coords)
+            yield pa.concat_tables(_arrow_table_reader(self.sr)), tuple(
+                self.joinids[0 : self.major_axis]
+                + [pa.array(coord_chunk)]
+                + self.joinids[self.major_axis + 1 : self.ndim]
+            )
```

Comment:
I think just switching out one element of a list would make this easier to understand:

```python
joinids = list(self.joinids)
joinids[self.major_axis] = pa.array(coord_chunk)
yield ..., tuple(joinids)
```